### PR TITLE
Add Fortinet SSL login scanner Aux Mod

### PR DESCRIPTION
**Pull request by maxmichels**
Adding a new scanner to bruteforce the login of Fortinet SSL VPN.

## Verification

List the steps needed to make sure this thing works

- [ ] Start `msfconsole`
- [ ] `use auxiliary/scanner/http/fortinet_ssl_vpn`
- [ ] set  username
- [ ] set  password
- [ ] set  rhost
- [ ] run

### CODE REVIEW

**Comment by mcmichels** [comment]
I dont understand the sanity test. i just removed unused lines and now it is failing?! when i execute the module on my computer it is working properly.

**Comment by bcoles** [comment]
> 
> 
> I dont understand the sanity test. i just removed unused lines and now it is failing?! when i execute the module on my computer it is working properly.

The failed sanity test can be safely ignore. The infrastructure goes insane every weekend so as to facilitate later contrast with normality.

**Comment by mcmichels** [comment]
Do i need to write doc before you merge? It is my first commit and i do not know the workflow exactly.. 🙈

File: modules/auxiliary/scanner/http/fortinet_ssl_vpn.rb [hunk 1]
```diff
@@ -0,0 +1,177 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'rex/proto/http'
```
Comment (jrobles-r7) [inline modules/auxiliary/scanner/http/fortinet_ssl_vpn.rb]: You may be able to remove this.
```suggestion
```

File: modules/auxiliary/scanner/http/fortinet_ssl_vpn.rb [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+      'DefaultOptions' =>
+        {
+          'SSL' => true,
+          'RPORT' => 443
+        }
+    ))
+
+    register_options(
+      [
+        Opt::RPORT(443),
```
Comment (jrobles-r7) [inline modules/auxiliary/scanner/http/fortinet_ssl_vpn.rb]:
```suggestion
```

You set `RPORT` in `DefaultOptions`.

**Comment by jrobles-r7** [comment]
Please add [module documentation](https://github.com/rapid7/metasploit-framework/blob/master/documentation/modules/module_doc_template.md).

File: modules/auxiliary/scanner/http/fortinet_ssl_vpn.rb (hunk 1 above)
Comment (mcmichels) [reply modules/auxiliary/scanner/http/fortinet_ssl_vpn.rb]: Totally right. Solved in commit.

File: modules/auxiliary/scanner/http/fortinet_ssl_vpn.rb (hunk 2 above)
Comment (mcmichels) [reply modules/auxiliary/scanner/http/fortinet_ssl_vpn.rb]: See new commit. Now solved (hopefully).

**Comment by mcmichels** [comment]
What are next steps to merge the module? Do I need to do something?

**Comment by acammack-r7** [comment]
@mcmichels This looks good, but it uses the old style of writing login scanners. If you feel up to it, we would prefer you take advantage of the newer tooling we have created for login scanners as shown in this guide: https://github.com/rapid7/metasploit-framework/wiki/How-to-write-a-HTTP-LoginScanner-Module#step-3-start-with-a-loginscanner-template

**Comment by mcmichels** [comment]
> If you feel up to it, we would prefer you take advantage of the newer tooling we have created for login scanners 

I will try to do it next week as my time this week is limited. Hope i can manage to do it. Otherwise i will commit the week after...

**Comment by wchen-r7** [comment]
Hi @mcmichels Apologies for not being familiar with the product. Is the server hardware or there's something I could download and test? Thank you.

**Comment by mcmichels** [comment]
I use hardware but there is also a vm. But you will need a valid license subscription to start the vm.

**Comment by wchen-r7** [comment]
Ok, in that case I won't be able to test. But it's okay, usually in that type of scenario we ask the author for a PCAP to show that the module is in working condition, and then land it. Hopefully it isn't too much trouble, but could you please email us a PCAP to msfdev[at]metasploit.com? In the subject you should just say "PCAP for PR 11427", thank you!

**Comment by jhart-r7** [comment]
As part of that PCAP would it also be possible to see the (presumed) default SSL certificate details and the complete HTTP response to the initial `GET /`?  Would be useful for exposure metrics/etc.

**Comment by mcmichels** [comment]
@jharms what do you mean exactly? Due i am totally new to metasploit and working on my first module the last 2 weeks i am not familiar with your processes.

**Comment by jhart-r7** [comment]
> @jharms what do you mean exactly? Due i am totally new to metasploit and working on my first module the last 2 weeks i am not familiar with your processes.

@mcmichels I was curious what the results of your `check_conn` method looks like when run against an applicable device.   It makes an HTTPS request to `/` and as part of that process an SSL certificate will be provided by the server and upon successful negotiation you'll get an HTTP response.  I was curious about the details of that SSL certificate and the HTTP response for potentially improving the `check_conn` method (by validating status code, body contents, etc) or for other general research efforts outside of metasploit.  

I suspect the PCAP that @wchen-r7 requested will have what need, and if it doesn't, no worries! Thank you for your contribution!

**Comment by mcmichels** [comment]
> HTTP response

This just checks for general responsibility of the system. Not for any specific answer nor ssl certificate. The check that it is a fortigate is made by is_app_ssl_vpn

If you think this is overhead i can remove that check and integrate the timeout check/responsibility into is_app_ssl_vpn

**Comment by wchen-r7** [comment]
@mcmichels It turns out Jon found what he was looking for, so we're good. You don't need to remove the check, that one is good. Just go ahead and email us the PCAP and we should be good to go. Sorry for the confusion!

**Comment by mcmichels** [comment]
@wchen-r7 okay, i will try to produce the pcap tomorrow.

**Comment by wchen-r7** [comment]
Thank you!

**Comment by mcmichels** [comment]
Sorry for the delay... i have just send the pcap to you.

**Comment by wchen-r7** [comment]
No problem. Thank you so much for the pcap. I'll start landing it now.

**Comment by wchen-r7** [comment]
# Release Notes

This adds a new login module for Fortinet SSL VPN server.

**Pull request merged**